src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataLatestIT.java

```diff
@@ -1050,6 +1084,40 @@ public void testUnderscoreInSchemaNamePatternForPrimaryAndForeignKeys() throws E
     }
   }
 
+  /**
+   * For driver versions higher than 3.14.5 we can disable the abilty to use pattern searches for
+   * getPrimaryKey and getForeignKey functions by setting enablePatternSearch = false.
+   */
+  @Test
+  public void testUnderscoreInSchemaNamePatternForPrimaryAndForeignKeysWithPatternSearchDisabled()
+      throws Exception {
+    Properties properties = new Properties();
+    properties.put(ENABLE_PATTERN_SEARCH, false);
+
+    try (Connection con = getConnection(properties);
+        Statement statement = con.createStatement()) {
+      String database = con.getCatalog();
+      TestUtil.withRandomSchema(
+          statement,
+          customSchema -> {
+            String escapedSchema = customSchema.replace("_", "\\_");
+            statement.execute("use schema " + customSchema);
+            statement.execute(
+                "create or replace table PK_TEST (c1 int PRIMARY KEY, c2 VARCHAR(10))");
+            statement.execute(
+                "create or replace table FK_TEST (c1 int REFERENCES PK_TEST(c1), c2 VARCHAR(10))");
+            DatabaseMetaData metaData = con.getMetaData();
+            // We have disabled the pattern search so we should get no results.
+            try (ResultSet rs = metaData.getPrimaryKeys(database, escapedSchema, null)) {
```

Comment:
we should add 2 more cases where we pass unescaped schema and number of keys in both situations is 1?